syntax = "proto3";

package xiaoai;

// 小艾服务接口定义
service XiaoAIService {
  // 整合四诊数据
  rpc IntegrateFourDiagnosis(IntegrateDiagnosisRequest) returns (IntegrateDiagnosisResponse) {}
  
  // 分析体质
  rpc AnalyzeConstitution(ConstitutionAnalysisRequest) returns (ConstitutionAnalysisResponse) {}
}

// 整合四诊数据请求
message IntegrateDiagnosisRequest {
  string inquiry_data = 1;       // 问诊数据（JSON字符串）
  string look_data = 2;          // 望诊数据（JSON字符串）
  string listen_data = 3;        // 闻诊数据（JSON字符串）
  string palpation_data = 4;     // 切诊数据（JSON字符串）
}

// 整合四诊数据响应
message IntegrateDiagnosisResponse {
  string integration_result = 1; // 整合结果（JSON字符串）
  float confidence = 2;          // 置信度
}

// 体质分析请求
message ConstitutionAnalysisRequest {
  string user_id = 1;            // 用户ID
  string symptoms = 2;           // 症状列表（JSON字符串）
  string patterns = 3;           // 证型列表（JSON字符串）
}

// 体质分析响应
message ConstitutionAnalysisResponse {
  string analysis_result = 1;    // 分析结果（JSON字符串）
  float confidence = 2;          // 置信度
}